statistics/dbstats_test.go

```diff
@@ -0,0 +1,533 @@
+	"github.com/letsencrypt/boulder/test"
+	"github.com/letsencrypt/boulder/test/vars"
+)
+
+var log = blog.UseMock()
+var ctx = context.Background()
+var certCounter int64
+
+type TestLogger struct {
+	*testing.T
+}
+
+func (testLog *TestLogger) Printf(format string, args ...interface{}) {
+	testLog.Logf(format, args)
+}
+
+// initSA constructs a r/w SQLStorageAuthority and a clean up function
+// that should be defer'ed to the end of the test.
+func initSA(t *testing.T) (*gorp.DbMap, *sa.SQLStorageAuthority, clock.FakeClock, func()) {
+	dbMap, err := sa.NewDbMap(vars.DBConnSA, 0)
+	test.AssertNotError(t, err, "Should not error")
+
+	fc := clock.NewFake()
+	fc.Set(time.Date(2015, 3, 4, 5, 0, 0, 0, time.UTC))
+
+	sa, err := sa.NewSQLStorageAuthority(dbMap, fc, log)
+	test.AssertNotError(t, err, "Should not error")
+
+	if testing.Verbose() && !testing.Short() {
+		dbMap.TraceOn("r/w", &TestLogger{t})
+	}
+
+	cleanUp := test.ResetSATestDatabase(t)
+	return dbMap, sa, fc, cleanUp
+}
+
+func parseConfigDuration(durationString string) cmd.ConfigDuration {
+	duration, _ := time.ParseDuration(durationString)
+	return cmd.ConfigDuration{Duration: duration}
+}
+
+func updateOCSP(t *testing.T, rwMap *gorp.DbMap, serial string, updateDate time.Time) {
+	_, err := rwMap.Exec(
+		`UPDATE certificateStatus
+     SET ocspResponse=?,ocspLastUpdated=?
+     WHERE serial=?`,
+		[]byte{},
+		updateDate,
+		serial,
+	)
+	test.AssertNotError(t, err, "Should not error")
+}
+
+func addCertificate(t *testing.T, rwMap *gorp.DbMap, saObj *sa.SQLStorageAuthority, ctx context.Context, regID int64, issueDate time.Time) *big.Int {
+	testKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	test.AssertNotError(t, err, "Should not error")
+
+	expiry := issueDate.AddDate(0, 0, 7)
+	serial := big.NewInt(certCounter)
+	certCounter += 1
+	rawCert := x509.Certificate{
+		Subject: pkix.Name{
+			CommonName: "example.com",
+		},
+		NotBefore:    issueDate,
+		NotAfter:     expiry,
+		DNSNames:     []string{"example-a.com"},
+		SerialNumber: serial,
+	}
+
+	certDER, err := x509.CreateCertificate(rand.Reader, &rawCert, &rawCert, &testKey.PublicKey, testKey)
+	test.AssertNotError(t, err, "Should not error")
+
+	_, err = saObj.AddCertificate(ctx, certDER, regID)
+	test.AssertNotError(t, err, "Should not error")
+
+	updateOCSP(t, rwMap, core.SerialToString(serial), issueDate)
+	return serial
+}
+
+func addRegistration(t *testing.T, sa core.StorageAdder, date time.Time) core.Registration {
+	contact, err := core.ParseAcmeURL("mailto:foo@example.com")
+	test.AssertNotError(t, err, "unable to parse contact link")
+
+	contacts := []*core.AcmeURL{contact}
+	privKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	test.AssertNotError(t, err, "unable to generate key")
+
+	reg, err := sa.NewRegistration(context.Background(), core.Registration{
+		Key:       jose.JsonWebKey{Key: privKey.Public()},
+		Contact:   contacts,
+		InitialIP: net.ParseIP("88.77.66.11"),
```

Comment:
small nit: is this particular IP meaningful? Could we use something from a range reserved for documentation in RFC5737?